### PR TITLE
add e2e test on memory cost

### LINKED ISSUE #71: e2e: Test Memory cost model
cc @islinwb

### PR DESCRIPTION
**Pull request by islinwb**
Fixes #71

### CODE REVIEW

**Comment by m1093782566** [comment]
@islinwb Thanks for your PR, but I would suggest combine this test with CPU cost test.

**Comment by islinwb** [comment]
@m1093782566

**Comment by m1093782566** [comment]
Given the huge code duplication with CPU test model, can we merge this test case into 

`should validates resource limits of pods that are allowed to run`?

**Comment by islinwb** [comment]
@m1093782566 OK.

File: test/e2e/poseidon_integration.go [hunk 1]
```diff
@@ -407,6 +407,111 @@ var _ = Describe("Poseidon", func() {
...
+					framework.Logf("Pod %v requesting resource cpu=%vm, memory=%vm on Node %v", pod.Name, getRequestedCPU(pod), getRequestedMemory(pod), pod.Spec.NodeName)
+					nodeToAllocatableMemMap[pod.Spec.NodeName] -= getRequestedMemory(pod)
+				}
+			}
+
+			By("Starting Pods to consume most of the cluster Memory.")
+			// Create one pod per node that requires 70% of the node remaining Memory.
+			fillerPodsMem := []*v1.Pod{}
+			for nodeName, memory := range nodeToAllocatableMemMap {
+				requestedMemory := memory * 3 / 10
```
Comment (shivramsrivastava) [inline test/e2e/poseidon_integration.go]: @islinwb The comment in line ```456``` says ```70%``` of the node's memory will be used but here you are using only ```30%``` of the remaining memory from the nodes.

**Comment by shivramsrivastava** [comment]
@islinwb 
I checked the logs on the previous failures.
It looks like even a filler-pod with ```50%``` was not running on the nodes because of ```Out of Memory``` exception.  But looks like ```Firmament``` scheduled that pod successfully.

It looks like your test case has found a real **bug** in the cpu-mem cost model.
It would be great if we can fix the bug rather than making this test case pass.

Some pointer to work on this bug.

1) Check if the node mem utilization calculation is correct in the test case.
2) Use the old ```50%``` mem configuration for the filler pods and see if it fails again.
3) Is the fails again, we need to debug on the ```Firmament``` container with more verbose logs.
    If no logs are there, please add them.

The problem is since we are not using heapster to feed in the real cluster utilization and we do some static calculation on the firmament, which is based only on the pods scheduled by ```Firmament```.
So what ```Firmament``` does, in this case, is it keeps reducing the mem and CPU for the pods it scheduled from the total allocatable memory which it received at the time of node addition.

We did couple of bug fixes on the cost calculation with node-affinity changes, but it always better to be sure.

**Comment by islinwb** [comment]
@shivramsrivastava Thanks! I'll ll look into it.

**Comment by islinwb** [comment]
@m1093782566 @shivramsrivastava 
It turns out the error was caused by the formatting of a quantity. I mistatkenly used the CPU's.

**Comment by m1093782566** [comment]
Oh! Really?

/hold cancel

**Comment by shivramsrivastava** [comment]
Ah, that was my hunch too.

**Pull request merged**